src/providers/Web3Provider.tsx

```diff
@@ -121,9 +182,9 @@ export const Web3ContextProvider: FC<PropsWithChildren> = ({ children }) => {
   }
 
   const isMetaMaskInstalled = async () => {
-    const provider = await detectEthereumProvider();
+    const provider = await detectEthereumProvider()
 
-    return !!window.ethereum && provider === window.ethereum;
+    return !!window.ethereum && provider === window.ethereum
   }
 
   const connectWallet = async () => {
```

Comment:
```ts
const [account] = await window.ethereum.request?.({ method: 'eth_requestAccounts' })`
```

This looks wrong.
- if window.ethereum is missing, it will throw `Cannot read properties of undefined`
- if window.ethereum.request is missing, it will throw `undefined is not iterable`

I'd expect
```tsx
    const accounts = await window.ethereum?.request?.({ method: 'eth_requestAccounts' })

    if (!accounts) {
      throw new Error('[Web3Context] Request account failed!')
    }

    await _init(accounts[0])
```

Reply:
Sure, `isMetaMaskInstalled` method was intended to check for that case, if `window.ethereum` wouldn't be defined, or would not be a "correct" wallet object. But since I added the option to skip, it is a valid point 👍